src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs

```diff
@@ -10,56 +10,25 @@
 
 namespace Dynamo.ViewModels
 {
-    /// <summary>
-    /// Proxy port view model, used for proxy ports under collapsed groups
-    /// Certain features could behave differently e.g. Node AutoComplete is 
-    /// disabled for it.
-    /// </summary>
-    public class ProxyPortViewModel : PortViewModel
-    {
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="node"></param>
-        /// <param name="port"></param>
-        public ProxyPortViewModel(NodeViewModel node, PortModel port)
-            : base(node, port)
-        {
-            // Do nothing for now.
-        }
-    }
-
     /// <summary>
     /// Port View Model
     /// </summary>
     public partial class PortViewModel : ViewModelBase
     {
         #region Properties/Fields
 
-        private readonly PortModel _port;
-        private readonly NodeViewModel _node;
+        protected readonly PortModel _port;
+        protected readonly NodeViewModel _node;
         private DelegateCommand _useLevelsCommand;
         private DelegateCommand _keepListStructureCommand;
-        private DelegateCommand _breakConnectionsCommand;
-        private DelegateCommand _hideConnectionsCommand;
+        private bool _showUseLevelMenu;
         private const double autocompletePopupSpacing = 2.5;
-        private SolidColorBrush portBorderBrushColor = new SolidColorBrush(Color.FromArgb(255, 204, 204, 204));
-        private SolidColorBrush portValueMarkerColor = new SolidColorBrush(Color.FromArgb(255, 204, 204, 204));
-        private SolidColorBrush portBackgroundColor = new SolidColorBrush(Color.FromArgb(0, 60, 60, 60));
         internal bool inputPortDisconnectedByConnectCommand = false;
-        private bool _showUseLevelMenu;
-        private bool areConnectorsHidden;
-        private string showHideWiresButtonContent = "";
-        private bool hideWiresButtonEnabled;
+        protected static SolidColorBrush portBorderBrushColor = new SolidColorBrush(Color.FromArgb(255, 204, 204, 204));
```

Comment:
appears some of these are repeated in the other new classes...

Reply:
There are other colors but they are specific for inputs not dup